src/txn/txn_ckpt.c

```diff
@@ -956,6 +962,31 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
 
     __checkpoint_verbose_track(session, "sync completed");
 
+    /* If the history store file exists on disk, update its statistic. */
+    WT_ERR(__wt_metadata_search(session, WT_METAFILE_URI, &config));
```

Comment:
I think we don't need to search the metadata to get whether we have a hs or not. Just check whether hs_dhandle is not null is enough.